observer/observer.go

```diff
@@ -0,0 +1,68 @@
+package observer
+
+import (
+	"strconv"
+
+	blog "github.com/letsencrypt/boulder/log"
+	"github.com/letsencrypt/boulder/metrics"
+
+	// _ are probes imported to trigger init func
+	_ "github.com/letsencrypt/boulder/observer/probes/dns"
+	_ "github.com/letsencrypt/boulder/observer/probes/http"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	statMonitors = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "obs_monitors",
+			Help: "count of configured monitors",
+		},
+		[]string{"name", "type", "valid"},
+	)
+	statObservations = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "obs_observations",
+			Help:    "time taken for a monitor to perform a request/query",
+			Buckets: metrics.InternetFacingBuckets,
+		},
+		[]string{"name", "type", "result"},
+	)
+)
+
+// Observer contains the parsed, normalized, and validated configuration
+// describing a collection of monitors and the metrics to be collected
+type Observer struct {
+	Logger   blog.Logger
+	Metric   prometheus.Registerer
+	Monitors []*monitor
+}
+
+// Start registers global metrics and spins off a goroutine for each of
+// the configured monitors
+func (o Observer) Start() {
```

Comment:
It feels like a "Start" function like this could return an error if it fails to start. The failure to start might also be a good metric to store for alerting to a bad monitor.

Reply:
I think the only errors that can happen here are panics.